Framework/CurveFitting/src/Algorithms/LeBailFit.cpp

```diff
@@ -446,15 +447,15 @@ void LeBailFit::processInputBackground() {
 
 //----------------------------------------------------------------------------------------------
 /** Calcualte LeBail diffraction pattern:
- *  Output spectra:
- *  0: data;  1: calculated pattern; 3: difference
- *  4: input pattern w/o background
- *  5~5+(N-1): optional individual peak
- */
+*  Output spectra:
+*  0: data;  1: calculated pattern; 3: difference
+*  4: input pattern w/o background
+*  5~5+(N-1): optional individual peak
+*/
 void LeBailFit::execPatternCalculation() {
   // Generate domain and values vectors
-  const MantidVec &vecX = m_dataWS->readX(m_wsIndex);
-  MantidVec &vecY = m_outputWS->dataY(CALDATAINDEX);
+  const auto &vecX = m_dataWS->x(m_wsIndex).rawData();
+  std::vector<double> vecY(m_outputWS->y(CALDATAINDEX).size(), 0);
```

Comment:
Having read a bit more below: You had to get into some mess to get the new data types into `calculateDiffractionPattern`. Have you considered changing the parameter types of the method to `HistogramX` and `HistogramY` instead of `std::vector<double>`?  I think that would make things much easier at the call sites.